includes/FailedRequestQueue/FailedRequest.php

```diff
@@ -80,6 +80,16 @@ public function incrementFailedAttempts(): void
         $this->failedAttempts++;
     }
 
+    /**
+     * Marks the failed job to not be retried again.
+     *
+     * @return void
+     */
+    public function doNotRetry(): void
+    {
+        $this->failedAttempts = self::MAX_ATTEMPTS;
```

Comment:
would it make sense in some scenario to mark it with e.g self::FAILED_REQUEST (represents a value of e.g. -1) to make it distinguishable from a 3-times-retried request?

Reply:
We currently do not have a sceneraio where we need to distinguish between these cases.

Reply:
fair enough